### DISCUSSION TITLE
ERROR - Razor Template not found : config on Payment screen

### DISCUSSION THREAD

**Discussion by alex-pommier** [openstore-ecommerce/OpenStore#294] [Q&A] [answered]
Hello,

I'm getting the following error on the payment section of the checkout process. Can anyone shed some light on what the issue might be? The payment to Paypal works fine when clicking the button.

ERROR - Razor Template Not Found: config.

![ARError](https://github.com/openstore-ecommerce/OpenStore/assets/13565738/dd90efbc-77b8-4f7c-9ba3-e954079213ef)

**Comment by alex-pommier** [comment]
BUMP does anyone have any idea what this problem could be?

**Reply by jasclar** [reply]
I have a similar issue - still not fully resolved. It seems to be related to other payment methods that have not had their details filled in and saved, e.g. Manual payments. Unticking the "Active" boxes for the PayPal payment method in the Plugins settings may also help?

**Reply by alex-pommier** [reply]
Hi @jasclar 

Thank you for the reply! I was digging through the template files to try and find the line to hide that warning because the checkout works fine but that error message dissuades people from checking out. I'll go through and check all my methods and hopefully get it removed but did you find any way of simply hiding that line? It doesn't have any CSS selector so I can't even use CSS to hide it.

**Comment by alex-pommier** [comment]
Can someone from Open Store please provide some insights here we're going on 2 weeks with no reply and it seems like this issue is affecting others using the software as well.

Things i've tried:
Store Administration > Admin > Settings > Store Configuration > Skip payment page (Turned this setting on: made no change)
Store Administration > Admin > Plugins >  OS_PayPal Payment > Interfaces > (2 show up here I tried turning off 1 at a time made no change)

![accessreliability-com-Admin-Store-Administration](https://github.com/openstore-ecommerce/OpenStore/assets/13565738/8b4acc45-bcc5-4dd4-95fe-c9629e958fff)

![accessreliability-com-Admin-Store-Administratio2n](https://github.com/openstore-ecommerce/OpenStore/assets/13565738/608e49d1-a4f3-484b-a4c8-c35774d27710)

**Comment by DNNMonster** [comment]
Sorry Alex but I haven't used the skip payment feature so I'd have to do some exploring.  I am also not 100% sure on the template error but I'd take a look and verify you have the NBright/Providers/ManualPaymentProvider/Themes/config/default/methodselection.html file in place and the setting is also matching that filename.  You may also want to review the db PLUGIN type code entries just in case something is amiss.  Occasionally we've seen duplicate PLUGIN typecode entries with PortalIds of 99999 post plugin upgrade.  If you see a dup then you will need to remove this manually.

**Comment by alex-pommier** [comment]
@DNNMonster I did find dups in the "[NBrightBuy]" table with reference to OSPaypal so I deleted the one with PortalID 99999 but that didn't fix the issue. I checked for the methodselection.html file in the default & OS_Paypal directories and it does exist in both. Not sure what next steps are, can you please advise.

Alternatively if you can tell me where the code can be found that renders that string I can encapsulate it in a div with a class and just hide it via CSS since the functionality of the checkout works fine.

**Reply by DNNMonster** [reply, marked as answer]
Still seems like a template issue related to a missing file or bogus setting.  I think I may have sent a slightly wrong file name earlier with the .html extension.

Can you verify there is a NBright\NBrightBuy\Providers\ManualPaymentProvider\Themes\config\default\methodselection.cshtml?

Also can you review your Manual Payment settings so as to make sure that the template input is using the exact same filename with a matching file extension.  ie.  methodselection.cshtml

For a css work around...  Isn't that message already wrapped in a div with a "defaultmessage" class?  You can review the source from the browser to verify this but that should work for the css selector if you have no success solving the root problem.   

I'm still thinking the issue is in the Manual Payment provider settings or a missing file.  Makes the most sense but if the files and settings are right for the Manual Payment provider I'm a bit stymied.  I'm not able to reproduce it unless I mess up those settings or remove a file.

**Reply by alex-pommier** [reply]
@DNNMonster I did as you suggested and just copy/pasted the name into the manual payment config area of the backend and saved that seemed to reset/solve the issue. The filename was correct but maybe having deleted that record in SQL + saving triggered something that fixed the problem.

Just as a note, the error message was not wrapped individually to be able to target it via CSS.

Thanks for the help!
Alex
